src/Sylius/Bundle/CoreBundle/Fixture/Factory/PromotionRuleExampleFactory.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Fixture\Factory;
+
+use Sylius\Component\Core\Factory\PromotionRuleFactoryInterface;
+use Sylius\Component\Core\Promotion\Checker\Rule\ContainsProductRuleChecker;
+use Sylius\Component\Core\Promotion\Checker\Rule\CustomerGroupRuleChecker;
+use Sylius\Component\Core\Promotion\Checker\Rule\HasTaxonRuleChecker;
+use Sylius\Component\Core\Promotion\Checker\Rule\NthOrderRuleChecker;
+use Sylius\Component\Core\Promotion\Checker\Rule\ShippingCountryRuleChecker;
+use Sylius\Component\Core\Promotion\Checker\Rule\TotalOfItemsFromTaxonRuleChecker;
+use Sylius\Component\Promotion\Checker\Rule\CartQuantityRuleChecker;
+use Sylius\Component\Promotion\Checker\Rule\ItemTotalRuleChecker;
+use Sylius\Component\Promotion\Model\PromotionRuleInterface;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class PromotionRuleExampleFactory extends AbstractExampleFactory implements ExampleFactoryInterface
+{
+    /**
+     * @var PromotionRuleFactoryInterface
+     */
+    private $promotionRuleFactory;
+
+    /**
+     * @var \Faker\Generator
+     */
+    private $faker;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $optionsResolver;
+
+    /**
+     * @param PromotionRuleFactoryInterface $promotionRuleFactory
+     */
+    public function __construct(PromotionRuleFactoryInterface $promotionRuleFactory)
+    {
+        $this->promotionRuleFactory = $promotionRuleFactory;
+
+        $this->faker = \Faker\Factory::create();
+        $this->optionsResolver = new OptionsResolver();
+
+        $this->configureOptions($this->optionsResolver);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $options = [])
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        /** @var PromotionRuleInterface $promotionRule */
+        $promotionRule = $this->promotionRuleFactory->createNew();
+        $promotionRule->setType($options['type']);
+        $promotionRule->setConfiguration($options['configuration']);
+
+        return $promotionRule;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setDefault('type', CartQuantityRuleChecker::TYPE)
+            ->setAllowedValues('type', [
```

Comment:
This effectively locks us to built-in rules, same as for actions. I'd love to use these fixtures to configure my custom promotions too. (as we do not have validation of the config, this is not a benefit at all)